### PR TITLE
RYA-148 Fixed a bug where Joins would write the same binding set with…

### PR DESCRIPTION
**Pull request by kchilton2**
… different visibility orders depending on which child node the new result appeared on.

### CODE REVIEW

File: extras/rya.pcj.fluo/pcj.fluo.app/src/test/java/org/apache/rya/indexing/pcj/fluo/app/IterativeJoinTest.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+/**
+ * Tests the methods of {@link IterativeJoin}.
+ */
+@RunWith(Parameterized.class)
+public class IterativeJoinTest {
+
+    @Parameters
+    public static Collection<Object[]> data() {
+        return Arrays.asList(new Object[][] {
```
Comment (isper3at) [inline extras/rya.pcj.fluo/pcj.fluo.app/src/test/java/org/apache/rya/indexing/pcj/fluo/app/IterativeJoinTest.java]: this just feels.....odd

**Comment by isper3at** [comment]
:+1:

File: extras/rya.pcj.fluo/pcj.fluo.app/src/main/java/org/apache/rya/indexing/pcj/fluo/app/JoinResultUpdater.java [hunk 2]
```diff
@@ -420,10 +423,20 @@ public VisibilityBindingSet next() {
...
+                rightVisi = joinResult.getVisibility();
+            } else {
+                leftVisi = joinResult.getVisibility();
+                rightVisi = newResult.getVisibility();
+            }
+
             String visibility = "";
             final Joiner join = Joiner.on(")&(");
-            final String leftVisi = newResult.getVisibility();
-            final String rightVisi = joinResult.getVisibility();
```
Comment (amihalik) [inline extras/rya.pcj.fluo/pcj.fluo.app/src/main/java/org/apache/rya/indexing/pcj/fluo/app/JoinResultUpdater.java]: Consider adding ColumnVisibility.flatten() to simplify the visibility expression.

Comment (kchilton2) [inline extras/rya.pcj.fluo/pcj.fluo.app/src/main/java/org/apache/rya/indexing/pcj/fluo/app/JoinResultUpdater.java]: Will do.

Comment (kchilton2) [inline extras/rya.pcj.fluo/pcj.fluo.app/src/main/java/org/apache/rya/indexing/pcj/fluo/app/JoinResultUpdater.java]: Using flatten makes a bunch of unit tests fail, so I'd rather address that in a different ticket.

**Comment by amihalik** [comment]
Merged.

**Pull request closed without merging**